scripts/core/editor3/components/Editor3.jsx

```diff
@@ -79,20 +131,25 @@ export class Editor3 extends React.Component {
         return false;
     }
 
-    /** Handle the editor tab key pressed event */
-    onTab(e) {
-        const maxDepth = 4;
-
-        this.onChange(RichUtils.onTab(e, this.state.editorState, maxDepth));
+    /**
+     * @ngdoc method
+     * @name Editor3#componentWillUpdate
+     * @description Called before update, init the current editor rectangle
+     */
+    componentWillUpdate() {
+        this.editorRect = ReactDOM.findDOMNode(this.refs.editor).getBoundingClientRect();
```

Comment:
Does this have to be here, before the update? I was thinking it could go into the render method. That way, you get the rectangle after the editor is rendered, which is the most recent size.